tests/functional/states/test_kubernetes.py

```diff
@@ -0,0 +1,1426 @@
+def test_pod_present(kubernetes, pod, _pod_spec):
+    """
+    Test kubernetes.pod_present creates a pod
+    """
+    ret = kubernetes.pod_present(
+        name=pod["name"],
+        namespace=pod["namespace"],
+        spec=_pod_spec,
+        wait=True,
+    )
+
+    assert ret.result is True
+    assert ret.changes
+
+    # Comment out idempotent test for now
+    # TODO: The state module needs fixed to handle proper present functionality
+
+
+# def test_pod_present_idempotency(kubernetes, pod, _pod_spec):
+#     """
+#     Test kubernetes.pod_present is idempotent
+#     """
+#     ret = kubernetes.pod_present(
+#         name=pod["name"],
+#         namespace=pod["namespace"],
+#         spec=_pod_spec,
+#         wait=True,
+#     )
+
+#     assert ret.result is True
+#     assert "already exists" in ret.comment
+#     assert not ret.changes
+
+
+def test_pod_present_test_mode(kubernetes, pod, _pod_spec):
+    """
+    Test kubernetes.pod_present in test mode
+    """
+    ret = kubernetes.pod_present(
+        name=pod["name"],
+        namespace=pod["namespace"],
+        spec=_pod_spec,
+        test=True,
+    )
+
+    assert ret.result is None
+    assert not ret.changes
+
+
+@pytest.mark.parametrize("pod", [False], indirect=True)
+def test_pod_present_template_context(kubernetes, pod, pod_template):
+    """
+    Test kubernetes.pod_present with template_context
+    """
+    template_context = {
+        "name": pod["name"],
+        "namespace": pod["namespace"],
+        "image": "nginx:latest",
+        "labels": {"app": "test"},
+    }
+
+    ret = kubernetes.pod_present(
+        name=pod["name"],
+        namespace=pod["namespace"],
+        source=pod_template,
+        template="jinja",
+        template_context=template_context,
+        wait=True,
+    )
+
+    assert ret.result is True
+    assert ret.changes
+
+
+def test_pod_absent(kubernetes, pod):
+    """
+    Test kubernetes.pod_absent deletes a pod
+    """
+    ret = kubernetes.pod_absent(
+        name=pod["name"],
+        namespace=pod["namespace"],
+        wait=True,
+    )
+
+    assert ret.result is True
+    assert ret.changes["kubernetes.pod"]["new"] == "absent"
+
+
+def test_pod_absent_test_mode(kubernetes, pod):
+    """
+    Test kubernetes.pod_absent in test mode
+    """
+    ret = kubernetes.pod_absent(
+        name=pod["name"],
+        namespace=pod["namespace"],
+        test=True,
+    )
+
+    assert ret.result is None
+    assert not ret.changes
+
+
+def test_pod_absent_idempotency(kubernetes, pod):
+    """
+    Test kubernetes.pod_absent is idempotent
+    """
+    ret = kubernetes.pod_absent(name=pod["name"], namespace=pod["namespace"], wait=True)
+    assert ret.result is True
+
+    # Test idempotency
+    ret = kubernetes.pod_absent(name=pod["name"], namespace=pod["namespace"])
+    assert ret.result is True
+    assert "does not exist" in ret.comment
+    assert not ret.changes
```

Comment:
```suggestion
@pytest.mark.parametrize("pod", [False], indirect=True)
def test_pod_absent_idempotency(kubernetes, pod):
    """
    Test kubernetes.pod_absent is idempotent
    """
    ret = kubernetes.pod_absent(name=pod["name"], namespace=pod["namespace"])
    assert ret.result is True
    assert "does not exist" in ret.comment
    assert not ret.changes
```

suggestion: Just try deleting a non-existent pod. I know that's stretching the definition of idempotency, but it's fine for our practical case.

This is valid for all `absent` idempotency tests in this file.